doc/sailfish-webview-webview.qdoc

```diff
@@ -14,4 +14,60 @@
   import namespace.
 */
 
+/*!
+  \qmlproperty var WebView::popupComponents
+  \brief Contains a dictionary of custom user interaction components
+
+  Client applications can define their own custom user interaction components
+  which extend the various interface types defined in the
+  \l{Sailfish.WebView.Popups} namespace.
+
+  In order for those custom components to be used by the WebView, the client
+  must then register those components with the WebView via this property.
+  The value is a dictionary whose keys are the asynchronous message types
+  which trigger the popup, and whose values are the \c{Component}s (or
+  component type definition URLs) which the WebView should load and display
+  when it receives the trigger message.
+
+  The dictionary keys are strings, and the values are either a custom
+  component entry object, or a popup-sub-type dictionary (whose key is
+  the sub-type name, and whose values are custom component entry objects).
+
+  The valid dictionary keys are as follows:
+  \list
+  \li \c{"Content:ContextMenu"} to trigger a \l{ContextMenuInterface}{context menu popup}
```

Comment:
```
    enum CustomPopup {
        ContextMenu,
        Alert,
        Confirm,
        ...
        GeoLocationPermission
    };
    Q_ENUM(CustomPopup)
```

That RawWebView is mostly an implementation detail. Hence, maybe some popup customisation related class.

Reply:
JavaScript dictionaries can only use strings as keys (at least in the ECMA version supported by our QML engine version), unfortunately...  I can provide the enum and then a custom lookup method which returns a string, I suppose.

Reply:
Even that doesn't work, unfortunately.  Within the JSON declaration of the dictionary object, you cannot resolve external symbols.  It's strings or nothing, if we want to keep the API user-friendly from QML, and retain this type of API (dictionary of components).

Reply:
That said, there's not too much difference between an enum and a string, except that for an enum you get some run-time check that you typed it correctly.  I'd personally not be averse to simply defining and documenting string keys... I will update the PR to do this, and you can take a look.

Reply:
Implemented now PopupProvider type instead of a raw JSON dict, so that these can be provided "nicely".  Seems to work well.  Documented etc.